examples_tests/50.MeshPacking/main.cpp

```diff
@@ -0,0 +1,190 @@
+    device->getCursorControl()->setVisible(false);
+
+    //! Since our cursor will be enslaved, there will be no way to close the window
+    //! So we listen for the "Q" key being pressed and exit the application
+    QToQuitEventReceiver receiver;
+    device->setEventReceiver(&receiver);
+
+    auto* driver = device->getVideoDriver();
+    auto* smgr = device->getSceneManager();
+    auto* am = device->getAssetManager();
+    auto* fs = am->getFileSystem();
+
+    //
+    auto* qnc = am->getMeshManipulator()->getQuantNormalCache();
+    //loading cache from file
+    qnc->loadNormalQuantCacheFromFile<asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10>(fs, "../../tmp/normalCache101010.sse", true);
+
+    // register the zip
+    device->getFileSystem()->addFileArchive("../../media/sponza.zip");
+
+    asset::IAssetLoader::SAssetLoadParams lp;
+    auto meshes_bundle = am->getAsset("sponza.obj", lp);
+    assert(!meshes_bundle.isEmpty());
+    auto mesh = meshes_bundle.getContents().begin()[0];
+    auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
+
+    //saving cache to file
+    qnc->saveCacheToFile(asset::CQuantNormalCache::E_CACHE_TYPE::ECT_2_10_10_10, fs, "../../tmp/normalCache101010.sse");
+
+    core::vector<ICPUMeshBuffer*> meshBuffers;
+    for (uint32_t i = 0u; i < mesh_raw->getMeshBufferCount(); i++)
+        meshBuffers.push_back(mesh_raw->getMeshBuffer(i));
+
+    DrawIndexedIndirectInput mdiCallParams;
+    SVertexInputParams vtxInputParams;
+    packMeshBuffers(driver, meshBuffers, vtxInputParams, mdiCallParams);
+
+    core::smart_refctd_ptr<IGPURenderpassIndependentPipeline> gpuPipeline;
+    setPipeline(driver, am, vtxInputParams, gpuPipeline);
+
+    //! we want to move around the scene and view it from different angles
+    scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeFPS(0, 100.0f, 0.5f);
+
+    camera->setPosition(core::vector3df(-4, 0, 0));
+    camera->setTarget(core::vector3df(0, 0, 0));
+    camera->setNearValue(1.f);
+    camera->setFarValue(5000.0f);
+
+    smgr->setActiveCamera(camera);
+
+    uint64_t lastFPSTime = 0;
+    while (device->run() && receiver.keepOpen())
+    {
+        driver->bindGraphicsPipeline(gpuPipeline.get());
+
+        driver->beginScene(true, true, video::SColor(255, 0, 0, 255));
+
+        //! This animates (moves) the camera and sets the transforms
+        camera->OnAnimate(std::chrono::duration_cast<std::chrono::milliseconds>(device->getTimer()->getTime()).count());
+        camera->render();
+
+        driver->pushConstants(gpuPipeline->getLayout(), asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(core::matrix4SIMD), camera->getConcatenatedMatrix().pointer());
+        driver->drawIndexedIndirect(mdiCallParams.vtxBindings, mdiCallParams.mode, mdiCallParams.indexType, mdiCallParams.indexBuff.get(), mdiCallParams.indirectDrawBuff.get(), mdiCallParams.offset, mdiCallParams.maxCount, mdiCallParams.stride);

```

Comment:
to draw everything with one drawcall you'd need to incorporate bits of ex 21 (megatexture) into all this to make sure all textures are accesible.